grid-proxy/pkg/client/grid_client.go

```diff
@@ -219,17 +220,55 @@ func (g *Clientimpl) NodeStatus(nodeID uint32) (res types.NodeStatus, err error)
 
 // Counters return statistics about the grid
 func (g *Clientimpl) Counters(filter types.StatsFilter) (res types.Counters, err error) {
-	query := statsParams(filter)
-	req, err := http.Get(g.url("stats%s", query))
+	response, err := g.makeRequest("stats", filter)
 	if err != nil {
-		return
+		return types.Counters{}, err
 	}
-	if req.StatusCode != http.StatusOK {
-		err = parseError(req.Body)
-		return
-	}
-	if err := json.NewDecoder(req.Body).Decode(&res); err != nil {
+
+	if err := json.NewDecoder(response.Body).Decode(&res); err != nil {
```

Comment:
I think no need to check error we can return directly as it's the end of the function